tests/test_default.py

```diff
@@ -0,0 +1,21 @@
+def test_brew_taps(Command):
+    c = Command('/usr/local/bin/brew tap --list')
+    assert c.rc == 0
+    tapped = c.stdout.split()
+    assert 'homebrew/core' in tapped
```

Comment:
@elliotweiser - But a developer shouldn't ever touch the roles themselves—they should be managed via `requirements.yml` or in some other way as a dependency. It's one case when designing software, another when dealing with infra. I'm working on a longer form post on why I believe this to be the case, but the gist of it is: since I maintain all these roles, and keep things as simple/efficient as possible, I don't want to maintain a separate set of tests that forms the 'contract' for the roles. I know what I want them to do, and that's enough for me :)

If the roles were maintained by a larger team, or if they were large enough that I couldn't fit the entire problem space in my head when working on them, I would be more prone to adding functional/system tests _on top of_ the built-in test that running the role provides, but for this role (and almost all the rest of my Ansible roles), they're simple enough that I could recite the list of tasks and includes almost by heart :)

Tests require maintenance, even in simpler cases, and for a guy maintaining over 150 projects on GitHub, and dozens more elsewhere, I only add tests where they are essential to my ability to continue maintaining the roles over a long period of time.

Anyways, I guess what I'm saying here is that I don't disagree with you, but in this case I don't want to merge in any extra test layers for this role (e.g. molecule, plus the extra test kitchen stuff).

And I would still like to have this PR split up further—can you break each change into it's own PR?